### DISCUSSION TITLE
Is it possible to use parallel computation in MibS?

### DISCUSSION THREAD

**Discussion by AprajayV** [coin-or/MibS#100] [Q&A]
Is it possible to build a version of mibs that can take advantage of the parallelization capability of ALPS. I tried building a mibs with --enable-static --disable-shared --with-mpi-cflags="$\(pkg-config --cflags mpi\)" --with-mpi-lflags="$\(pkg-config --libs mpi\)" MPICC=mpicc MPICXX=mpiCC.
I ran an example with Alps_processNum 4  in mibs.par, however on inspecting with top, I found out only one CPU is being engaged.
Is it possible to use parallel computation, if so what is the correct procedure?

**Comment by tkralphs** [comment]
Hi @AprajayV, sorry for the delay in replying. In fact, it is not yet possible to parallelize the tree search with MibS. This should probably not be too difficult to enable (since, as you note, MibS is built on top of a parallel search framework). In fact, it may even be straightforward, it just hasn't surfaced on my TODO list yet. In the meantime, a perhaps better way to utilize parallel resources is simply to parallelize the solution process of the underlying MILP solver. Depending on how difficult the second-level MILPs actually are, this may result in an even better speedup.
